src/components/stepforms/AddTotalRowsStepForm.vue

```diff
@@ -0,0 +1,123 @@
+      name="Columns to aggregate:"
+      v-model="aggregations"
+      :defaultItem="defaultAggregation"
+      :widget="widgetAggregation"
+      :automatic-new-field="false"
+      data-path=".aggregations"
+      :errors="errors"
+    />
+    <MultiselectWidget
+      class="groupsInput"
+      v-model="editedStep.groups"
+      name="(Optional) Group by:"
+      :options="columnNames"
+      @input="setSelectedColumns({ column: editedStep.groups[editedStep.groups.length - 1] })"
+      placeholder="Add columns"
+      data-path=".groups"
+      :errors="errors"
+    />
+    <StepFormButtonbar />
+  </div>
+</template>
+
+<script lang="ts">
+import { Prop } from 'vue-property-decorator';
+
+import { StepFormComponent } from '@/components/formlib';
+import TotalsWidget from '@/components/stepforms/widgets/Totals.vue';
+import { AddTotalRowsStep, AggFunctionStep } from '@/lib/steps';
+
+import BaseStepForm from './StepForm.vue';
+import AggregationWidget from './widgets/Aggregation.vue';
+import ListWidget from './widgets/List.vue';
+import MultiselectWidget from './widgets/Multiselect.vue';
+
+@StepFormComponent({
+  vqbstep: 'totals',
+  name: 'totals-step-form',
+  components: {
+    ListWidget,
+    MultiselectWidget,
+  },
+})
+export default class AddTotalRowsStepForm extends BaseStepForm<AddTotalRowsStep> {
+  @Prop({ type: Object, default: () => ({ name: 'totals', totals: [], aggregations: [] }) })
+  initialStepValue!: AddTotalRowsStep;
+
+  readonly title: string = 'Add Total Rows';
+  totalsWidget = TotalsWidget;
+  widgetAggregation = AggregationWidget;
+
+  get defaultAggregation() {
+    const agg: AggFunctionStep = {
+      columns: [],
+      newcolumns: [],
+      aggfunction: 'sum',
+    };
+    return agg;
+  }
+
+  get aggregations() {
+    if (this.editedStep.aggregations.length) {
+      return this.editedStep.aggregations;
+    } else {
+      return [this.defaultAggregation];
+    }
+  }
+
+  set aggregations(newval) {
+    this.editedStep.aggregations = [...newval];
+  }
+
+  get totals() {
+    if (this.editedStep.totals.length) {
+      return this.editedStep.totals;
+    } else {
+      return [[]];
+    }
+  }
+
+  set totals(newval) {
+    this.editedStep.totals = [...newval];
+  }
+
+  submit() {
+    /**
+     * If different aggregations have to be performed on the same column, add a suffix
+     * to the automatically generated newcolumn name
+     */
+    const newcolumnOccurences: { [prop: string]: number } = {};
+    for (const agg of this.editedStep.aggregations) {
+      agg.newcolumns = [...agg.columns];
+      for (const c of agg.newcolumns) {
+        newcolumnOccurences[c] = (newcolumnOccurences[c] || 0) + 1;
+      }
+    }
+    for (const agg of this.editedStep.aggregations) {
+      for (let i = 0; i < agg.newcolumns.length; i++) {
+        if (newcolumnOccurences[agg.newcolumns[i]] > 1) {
+          agg.newcolumns.splice(i, 1, `${agg.newcolumns[i]}-${agg.aggfunction}`);
+        }
+      }
+    }
```

Comment:
I think it would have been easier to test and explain if this was extracted into a dedicated function, that would take the editedStep as input, and return the step with suffixed columns if necessary.
That way, unit testing that function is way easier than doing it in the component, and the only test you need for the component is to check that this function has been correctly called.